internal/resources/mock_client.go

```diff
@@ -0,0 +1,20 @@
+package resources
+
+import "net/url"
+
+type MockClient struct {
```

Comment:
I ended up not going with the testify mock code. If you'd prefer that way, I can change this.